pkg/schedule/config/config_provider.go

```diff
@@ -49,9 +50,9 @@ type SchedulerConfigProvider interface {
 	SetSchedulingAllowanceStatus(bool, string)
 	GetStoresLimit() map[uint64]StoreLimitConfig
 
-	IsSchedulerDisabled(string) bool
-	AddSchedulerCfg(string, []string)
-	RemoveSchedulerCfg(string)
+	IsSchedulerDisabled(types.CheckerSchedulerType) bool
+	AddSchedulerCfg(types.CheckerSchedulerType, []string)
```

Comment:
we can consider removing this interface.  here is some background:

- v2.x all scheduler configs are stored in the global config `Config.SchedulerConfig` 
- v3.x let the scheduler has it's own config, that's store in another key/value in etcd https://github.com/tikv/pd/pull/1733/

To ensure compatibility, both configurations are retained, and we can now clean up the old one. However, there is one problem that needs to be solved: the default scheduler state needs special handling.

Reply:
It is better to do it in a new pr. I filed an issue to track it. https://github.com/tikv/pd/issues/8474

Reply:
BTW, do you mean all this interface or these two function?